npm/src/language-service/language-service.ts

```diff
@@ -195,15 +199,16 @@ export class QSharpLanguageService implements ILanguageService {
   onDiagnostics(uri: string, version: number, diagnostics: IDiagnostic[]) {
     try {
       const code = this.code[uri];
-      if (code === undefined) {
-        log.error(`expected ${uri} to be in the document map`);
+      const empty = diagnostics.length === 0;
+      if (code === undefined && !empty) {
```

Comment:
Why does this error depend on there being some diagnostics now? A comment to clarify would help.